src/modules/Schedule/components/ColorPicker/hooks/useColorPicker.ts

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react'
+
+import { ScheduleColor } from '@/modules/Schedule/components/ColorPicker/constants'
+import { CourseCartItem, courseCartStore } from '@/store'
+
+export type ColorClassKey = Pick<
+  CourseCartItem,
+  'courseNo' | 'studyProgram' | 'academicYear' | 'semester' | 'abbrName' | 'color'
+>
+
+export function useColorPicker(classKey: ColorClassKey) {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null)
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+  const open = !!anchorEl
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+  const setColor = (scheduleColor: ScheduleColor) => {
+    courseCartStore.changeColor(classKey, scheduleColor)
```

Comment:
Can we also measure here using useLog to know which color is user preferred most?

Reply:
Done.
Please check if i use it properly

Reply:
Will do it tommorow!